components.md

```diff
@@ -0,0 +1,62 @@
+# Components
+
+This page provides details on the main components/entities of the ECP.
+
+## Director
+The Director is the main controller orchestrating the whole thing, an object which can 
```

Comment:
I'd make it more general, that you can have more than one director:
```suggestion
A Director manages a part or the entire setup orchestrating the actors and observers according to the needs of the experiment.
```

Reply:
I see the use case for having more than one coordinator,  but I imagined that the director is a singleton. Otherwise, we don't have a single source of truth for commands.
If you have two directors, and one tells an instrument to put Voltage to 100, and another to 5, what then?

Reply:
> If you have two directors, and one tells an instrument to put Voltage to 100, and another to 5, what then?

Then the instrument puts the voltage to 100 and with the next command to 5.

## My reasons for not only one director.
- We need regularly more than one director (two or more people working independently on different parts of the beamline).
- Sometimes we need access to the same VISA resource (stepper motor card controlling 6 motors), which is difficult, as only one program may open that resource. The separation of director and actor will allow to use the same card from different directors.
- I envison the director to be (potentially) a simple script (create director communication part, send commands to actors, read actors, loop the last two steps a few times, print results, done). Afterwards you use another script as director. Forcing a single director seems more difficult

## Possible solution
- We could allow to lock/reserve actors for a director, such that an actor refuses commands (all or just potentially harmful commands like setters) from other directors than the original one.

If we do not reach an agreement within a few comments, we should move it to its own issue, I'd say.

Reply:
> more people working _independently_ on _different_ parts 

OK, this i can understand, but not two directors "fighting" over control of one voltage (say). 
You raise a good point about a multichannel device with different channels for different users, that was also what I thought of later. 
Locking would again make that usecase impossible, though. We could lock individual Channels, but that would not make sense for all instruments.

Let's not restrict to one now, and I'll open an issue about how to deal with control contention.